### PR TITLE
Added platform collision for player! 

### PR DESCRIPTION
**Pull request by BGamber**
Allows jumping off platforms as well. Uses colliderect() to detect player's rect touching a platform's rect.

### CODE REVIEW

**Comment by BGamber** [comment]
Oops, I also changed variable names in this commit; I was using both underscore_naming and camelCase so I switched them over to the former.

**Comment by BGamber** [comment]
Hey @illiaChaban, mind 👀-ing this for me?

File: pygaman.py [hunk 1]
```diff
@@ -9,33 +9,45 @@ def __init__(self, x, y):
         super(Pygaman, self).__init__()
         self.x = x
         self.y = y
-        self.moveSpeed = 0
-        self.vertSpeed = 0
+        self.move_speed = 0
```
Comment (illiaChaban) [inline pygaman.py]: nice job!

File: pygaman.py [hunk 2]
```diff
@@ -9,33 +9,45 @@ def __init__(self, x, y):
...
         self.direction = 'right'
-        self.rect = self.run1.get_rect()
-        self.jumpCount = 0
-
-    def update(self, window):
-        self.x += self.moveSpeed
+        self.rect = pygame.Rect(self.x, self.y, 30, 40)
+        self.jump_count = 0
 
+    def update(self, window, platforms):
```
Comment (illiaChaban) [inline pygaman.py]: the beast!

File: pygaman.py [hunk 3]
```diff
@@ -9,33 +9,45 @@ def __init__(self, x, y):
...
         frame = counter % frames
         sprite_index = frame / (frames / 2)
-        if self.vertSpeed != 0:
+        if self.vert_speed != 0:
             if self.direction == 'right':
                 return self.jump
             elif self.direction == 'left':
                 return pygame.transform.flip(self.jump, True, False)
-        elif self.moveSpeed > 0:
+        elif self.move_speed > 0:
```
Comment (illiaChaban) [inline pygaman.py]: how did you do that???

File: pygaman.py [hunk 4]
```diff
@@ -44,11 +56,11 @@ def render(self, frames=20, counter=0):
                 return pygame.transform.flip(self.run_anim[0], True, False)
 
     def gravity(self, window):
-        if (self.y + self.rect.height) >= (window.height - self.vertSpeed) and self.vertSpeed > 0:
-            self.vertSpeed = 0
-            self.jumpCount = 0
+        if (self.y + self.rect.height) >= (window.height - self.vert_speed) and self.vert_speed > 0:
```
Comment (illiaChaban) [inline pygaman.py]: camelCase really sucks!!

File: pygaman.py [hunk 5]
```diff
@@ -130,27 +144,32 @@ def main():
...
+                player.move_speed = -3
             else:
-                player.moveSpeed = 0
-            if event.type == pygame.KEYDOWN and event.key == pygame.K_UP and player.jumpCount < 2:
-                player.vertSpeed = -7
-                player.jumpCount += 1
+                player.move_speed = 0
+            if event.type == pygame.KEYDOWN and event.key == pygame.K_UP and player.jump_count < 2:
+                player.vert_speed = -7
+                player.jump_count += 1
```
Comment (illiaChaban) [inline pygaman.py]: you're genius!!

**Pull request merged**